source_gen/lib/src/generator_for_annotation.dart

```diff
@@ -34,18 +35,36 @@ abstract class GeneratorForAnnotation<T> extends Generator {
   TypeChecker get typeChecker => new TypeChecker.fromRuntime(T);
 
   @override
-  Future<String> generate(LibraryReader library, BuildStep buildStep) async {
-    var elements = library.annotatedWith(typeChecker);
-    var allOutput = await Future.wait(elements.map((e) async =>
-        await generateForAnnotatedElement(e.element, e.annotation, buildStep)));
-    // TODO interleave comments indicating which element produced the output?
-    return allOutput.where((o) => o != null && o.isNotEmpty).join('\n');
+  FutureOr<String> generate(LibraryReader library, BuildStep buildStep) async {
+    var set = await new Stream.fromIterable(library.annotatedWith(typeChecker))
```

Comment:
Choose better names for these (you are shadowing it below, and it's generally just a bad name lol).

Also isn't this a reserved word?

Reply:
fixed